src/sensors/rev_encoder.hpp

```diff
@@ -1,16 +1,58 @@
+/**
+ * @file rev_encoder.hpp
+ * @brief The header file for the Rev Encoder firmware
+ * @date 2023-10-31
+ */
+
+#include <cmath>
 #include <FreqMeasureMulti.h>
 
 #ifndef REV_ENCODER_H
 #define REV_ENCODER_H
 
-class RevEncoder {
+/**
+ * @brief the class for the Rev Encoder
+ */
+class RevEncoder 
+{
 	private:
-		uint8_t pin;
+		/**
+		* @brief the pin number that the encoders signal pin is plugged into
+		*/
+		uint8_t inPin;
+		/**
+		* @brief Used to read rise time of the encoder
+		*/
 		FreqMeasureMulti freq;
+		/**
+		* @brief measure of current angle in ticks (1024 ticks = 1 rotation). When set to 1 individual readings are returned
+		*/
+		float ticks;
```

Comment:
Make sure your software doesn't reduce its data resolution by restricting it to 1024 integers like we discussed earlier. Just check what resolution it supports, and as long as you don't do any int casts you should be fine

Reply:
yes

Reply:
You have an int cast, see below review